### PR TITLE
Handle registry boards in processor

### PR DESCRIPTION
**Pull request by AbdulrhmnGhanem**
I tried three approaches to handle the registry boards:

1. clone the repo, checkout the hash, create an empty gitea repo, and push to gitea (the most complicated)

2. migrate the repo, clone it, and change the default branch, and push

3. checkout the hash in the processor (the cleanest)

I settled for making the processor handle the registry boards.

### CODE REVIEW

File: processor/src/queues.ts [hunk 1]
```diff
@@ -234,3 +251,29 @@ async function alreadyProcessed(
...
+}
+
+const registryBoards = await getRegistryBoards()
+
+function getRegistryHash(localGitDir: string) {
+  const repoFullName = localGitDir
+    .split('/')
+    .slice(-2)
+    .join('/')
+    .replace('.git', '')
```
Comment (kasbah) [inline processor/src/queues.ts:270]:
```suggestion
    .replace(/\.git$/, '')
```

File: processor/src/queues.ts [hunk 2]
```diff
@@ -180,7 +181,14 @@ async function sync(gitDir, checkoutDir) {
     .acquire(gitDir, async done => {
       log.info('Acquired sync lock for ', gitDir)
 
+      const registryHash = getRegistryHash(gitDir)
+
       if (await exists(checkoutDir)) {
+        if (registryHash) {
+          // no need to pull if we aren't going to use the latest commit
+          done()
+        }
```
Comment (kasbah) [inline processor/src/queues.ts:190]: But you are still pulling below, you are just releasing the lock early?

**Review by kasbah — CHANGES_REQUESTED** [review]

File: processor/src/queues.ts [hunk 3]
```diff
@@ -201,6 +209,15 @@ async function sync(gitDir, checkoutDir) {
           }
         })
         log.debug('Cloned into', checkoutDir)
+        if (registryHash) {
```
Comment (kasbah) [inline processor/src/queues.ts:212]:
```suggestion
        if (registryHash != null) {
```

File: processor/src/queues.ts [hunk 4]
```diff
@@ -180,7 +181,14 @@ async function sync(gitDir, checkoutDir) {
     .acquire(gitDir, async done => {
       log.info('Acquired sync lock for ', gitDir)
 
+      const registryHash = getRegistryHash(gitDir)
+
       if (await exists(checkoutDir)) {
+        if (registryHash) {
```
Comment (kasbah) [inline processor/src/queues.ts:187]:
```suggestion
        if (registryHash != null) {
```

File: processor/src/queues.ts (hunk 2 above)
Comment (AbdulrhmnGhanem) [reply processor/src/queues.ts]: Resolved in 68230b6.

**Pull request merged**